### PR TITLE
Fixed Pagination support 📄

### PR DESCRIPTION
**Pull request by mikegrima**
- Fixed for S3 (and other technologies with different request and response pagination markers)
- Added support for pagination where the request and response markers are different
- Also addresses #17

### CODE REVIEW

File: cloudaux/orchestration/aws/s3.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def get_lifecycle(bucket_name, **conn):
         rule_dict = {
             'id': rule['ID'],
             'status': rule['Status'],
-            'prefix': rule['Prefix'],
+            'prefix': rule.get('Prefix'),
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/s3.py]: Is rule['Prefix'] sometimes not a valid key?

File: cloudaux/orchestration/aws/s3.py [hunk 2]
```diff
@@ -323,7 +323,7 @@ def get_bucket(bucket_name, output='camelized', include_created=False, **conn):
         'analytics_configurations': get_bucket_analytics_configurations(bucket_name, **conn),
         'metrics_configurations': get_bucket_metrics_configurations(bucket_name, **conn),
         'inventory_configurations': get_bucket_inventory_configurations(bucket_name, **conn),
-        '_version': 4
+        '_version': 5
```
Comment (monkeysecurity) [inline cloudaux/orchestration/aws/s3.py]: Does the version need to be bumped?  We're not changing the format of the response?

File: cloudaux/aws/decorators.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def increase_delay(e):
     return decorator
 
 
-def paginated(response_key, pagination_marker="Marker"):
+def paginated(response_key, request_pagination_marker="Marker", response_pagination_marker="Marker"):
```
Comment (monkeysecurity) [inline cloudaux/aws/decorators.py]: Very nice.

File: cloudaux/orchestration/aws/s3.py (hunk 1 above)
Comment (mikegrima) [inline cloudaux/orchestration/aws/s3.py]: Apparently, as documented in #17 it is?

File: cloudaux/orchestration/aws/s3.py (hunk 2 above)
Comment (mikegrima) [inline cloudaux/orchestration/aws/s3.py]: Good point -- fixing.

**Pull request merged**